_api_app/app/Sites/SitesDataService.php

```diff
@@ -123,16 +123,31 @@ public function create($cloneFrom = null)
         $name = 'untitled-' . uniqid();
         $dir = $this->XML_SITES_ROOT . '/' . $name;
 
-        @mkdir($dir, 0777, true);
-
         if ($cloneFrom != null) {
             $src = $cloneFrom == '0' ? $this->XML_MAIN_ROOT : $this->XML_SITES_ROOT . '/' . $cloneFrom;
+            $name = 'copy-of-'.$cloneFrom;
+            $title = 'Main site';
```

Comment:
Please use string `Main site` as a php class constant as it is used more than once in this class.